T/Trilinos/build_tarballs.jl

```diff
@@ -67,4 +169,4 @@ dependencies = [
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"7.1.0")
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"7.2.0")
```

Comment:
Why this change?  We don't have 7.2, this goes straight to 8

Reply:
I was attempting to match Kokkos' tested compiler set. I can change it back.